src/test/CMakeLists.txt

```diff
@@ -244,15 +249,43 @@ set(
     ${FILESYSTEM_LIBRARY}
     ${Boost_SYSTEM_LIBRARY}
     ${Boost_THREAD_LIBRARY}
+    ${CMAKE_DL_LIBS}
 )
 
 # Build special sanitizer version of googletest
 include_directories(../../third_party/googletest/googletest/)
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 
+# Configure TestPlugin
+add_library(TestPlugin SHARED
+    utils/test_plugin.cpp
+    utils/test_plugin.hpp
+)
+# Prevent the linker under macOS from complaining about undefined methods
+if (APPLE)
+    if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+        target_link_libraries(TestPlugin PRIVATE "-undefined dynamic_lookup")
+    endif()
+endif()
+
+# Configure NonInstantiablePlugin
+add_library(NonInstantiablePlugin SHARED
+    utils/non_instantiable_plugin.cpp
+)
+# Prevent the linker under macOS from complaining about undefined methods
+if (APPLE)
+    if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+        target_link_libraries(NonInstantiablePlugin PRIVATE "-undefined dynamic_lookup")
+    endif()
+endif()
+
+# We define LIB_DIR to always load plugins from the correct directory for testing purposes
+add_definitions(-DLIB_DIR="${CMAKE_BINARY_DIR}/lib/")
```

Comment:
That's a pretty generic name for the dir where plugins are to be found. May `TEST_PLUGIN_DIR`?